File: binaryTreeRightSide.java
// Time Complexity : O(n)
// Space Complexity : O(1) , do we add queue space used?
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No issues, followed level order way

public class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        if(root==null)
            return new ArrayList<Integer>();
        
        List<Integer> result=new ArrayList<>();
        Queue<TreeNode> queue=new LinkedList<>();
        queue.add(root);
        while(!queue.isEmpty())
        {
            int size=queue.size();
            TreeNode curr=null;
            for(int i=0;i<size;i++)
            {
                curr=queue.poll();
                
                if(curr.left!=null)
                    queue.add(curr.left);
                if(curr.right!=null)
                    queue.add(curr.right);
            }
            
            result.add(curr.val);
        }
        
        return result;
    }
}

File: binary_tree_right_side.cpp
//Time Complexicity = O(n)
//Space Complexicity  =O(n)
//Runs successfully on the leetcode    
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    vector<int>result;
public:
    vector<int> rightSideView(TreeNode* root) {
        if(root == NULL)
            return{};
        dfs(root,0);
        return result;
        
    }
    void dfs(TreeNode* root, int level)
    {
        if(root == NULL)
            return;
        if(level == result.size())
            result.push_back(root->val);
        dfs(root->right,level+1);
        dfs(root->left,level+1);
    }
};